### PR TITLE
Optimize the monitor plugin for FlexibleEngine (prioritize using Agent to query cpu and mem)

### LINKED ISSUE #381: Optimize the monitor plugin for FlexibleEngine
Optimize the monitor plugin for FlexibleEngine (prioritize using Agent to query cpu and mem).

### PR DESCRIPTION
**Pull request by jinyangyang222**
fix #381

### CODE REVIEW

File: plugins/flexibleengine/src/main/java/org/eclipse/xpanse/orchestrator/plugin/flexibleengine/monitor/models/FlexibleEngineMonitorMetrics.java [hunk 1]
```diff
@@ -14,4 +14,8 @@ public class FlexibleEngineMonitorMetrics {
     public static final String CPU_UTILIZED = "cpu_util";
 
     public static final String MEM_UTILIZED = "mem_util";
-}
+
+    public static final String CPU_USAGEIZED = "cpu_usage";
+
+    public static final String MEM_USEDPERCENTIZED = "mem_usedPercent";
```
Comment (swaroopar) [inline plugins/flexibleengine/src/main/java/org/eclipse/xpanse/orchestrator/plugin/flexibleengine/monitor/models/FlexibleEngineMonitorMetrics.java:20]: var name can be MEM_USED_IN_PERCENTAGE

File: plugins/flexibleengine/src/main/java/org/eclipse/xpanse/orchestrator/plugin/flexibleengine/monitor/models/FlexibleEngineMonitorMetrics.java [hunk 2]
```diff
@@ -14,4 +14,8 @@ public class FlexibleEngineMonitorMetrics {
     public static final String CPU_UTILIZED = "cpu_util";
 
     public static final String MEM_UTILIZED = "mem_util";
-}
+
+    public static final String CPU_USAGEIZED = "cpu_usage";
```
Comment (swaroopar) [inline plugins/flexibleengine/src/main/java/org/eclipse/xpanse/orchestrator/plugin/flexibleengine/monitor/models/FlexibleEngineMonitorMetrics.java:18]: var name can be CPU_USAGE

File: plugins/flexibleengine/src/main/java/org/eclipse/xpanse/orchestrator/plugin/flexibleengine/monitor/utils/FlexibleEngineMonitorConverter.java [hunk 3]
```diff
@@ -91,6 +133,15 @@ public Map<String, MonitorResourceType> buildMonitorMetricUrls(DeployResource re
         return urlTypeMap;
     }
 
+    /**
+     * Extract the namespace from the response and add it to the set.
+     */
+    public Set<String> getResponseNamespaces(
```
Comment (swaroopar) [inline plugins/flexibleengine/src/main/java/org/eclipse/xpanse/orchestrator/plugin/flexibleengine/monitor/utils/FlexibleEngineMonitorConverter.java:139]: getResponseNamespaces -> does not seem to be right. If it is the namespace of the metrics. then we can rename it to getMetricsNamespaces.

**Review by swaroopar — CHANGES_REQUESTED** [review]
namings to be changed.

File: plugins/flexibleengine/src/main/java/org/eclipse/xpanse/orchestrator/plugin/flexibleengine/monitor/utils/FlexibleEngineMonitorConverter.java (hunk 3 above)
Comment (jinyangyang222) [reply plugins/flexibleengine/src/main/java/org/eclipse/xpanse/orchestrator/plugin/flexibleengine/monitor/utils/FlexibleEngineMonitorConverter.java:139]: ok,thank you,I will modify them.

File: plugins/flexibleengine/src/main/java/org/eclipse/xpanse/orchestrator/plugin/flexibleengine/monitor/utils/FlexibleEngineMonitorConverter.java [hunk 4]
```diff
@@ -102,9 +153,10 @@ public Map<String, MonitorResourceType> buildMonitorMetricUrls(DeployResource re
      */
     public Metric convertResponseToMetric(DeployResource deployResource,
                                           MonitorResourceType type,
-                                          ShowMetricDataResponse response) {
+                                          ShowMetricDataResponse response,
+                                          FlexibleEngineNameSpaceKind flexibleEngineNameSpaceKind) {
         Metric metric = new Metric();
```
Comment (swaroopar) [inline plugins/flexibleengine/src/main/java/org/eclipse/xpanse/orchestrator/plugin/flexibleengine/monitor/utils/FlexibleEngineMonitorConverter.java:158]: Metric class also has a new mandatory field called "unit". I see this is not being set in the FlexibleEngine plugin. 
Can you please check this? Ideally without this, the REST API method will not work at all.

**Review by swaroopar — CHANGES_REQUESTED** [review]
mandatory param is not set.

File: plugins/flexibleengine/src/main/java/org/eclipse/xpanse/orchestrator/plugin/flexibleengine/monitor/utils/FlexibleEngineMonitorConverter.java (hunk 4 above)
Comment (jinyangyang222) [reply plugins/flexibleengine/src/main/java/org/eclipse/xpanse/orchestrator/plugin/flexibleengine/monitor/utils/FlexibleEngineMonitorConverter.java:158]: I have made changes, for both cpu and mem, the unit is '%'.

**Review by swaroopar — APPROVED** [review]
looks good.

**Pull request merged**